diff --git a/config.def.h b/config.def.h
index ff56bc8..eec8bf2 100644
--- a/config.def.h
+++ b/config.def.h
@@ -34,6 +34,9 @@ static const char *colors[][3]      = {
     [SchemeLayout]  = { green,      black,  black },
 };
 
+/* staticstatus */
+static const int statmonval = 0;
+
 /* tagging */
 static const char *tags[] = { "1", "2", "3", "4", "5" };
 
diff --git a/dwm.c b/dwm.c
index 0845fec..be9c52f 100644
--- a/dwm.c
+++ b/dwm.c
@@ -286,7 +286,7 @@ static Cur *cursor[CurLast];
 static Clr **scheme;
 static Display *dpy;
 static Drw *drw;
-static Monitor *mons, *selmon;
+static Monitor *mons, *selmon, *statmon;
 static Window root, wmcheckwin;
 
 /* configuration, allows nested code to access above variables */
@@ -469,7 +469,7 @@ buttonpress(XEvent *e)
             arg.ui = 1 << i;
         } else if (ev->x < x + blw)
             click = ClkLtSymbol;
-        else if (ev->x > selmon->ww - statusw) {
+        else if (m == statmon && ev->x > selmon->ww - statusw) {
             x = selmon->ww - statusw;
             click = ClkStatusText;
 
@@ -922,7 +922,7 @@ drawbar(Monitor *m)
         return;
 
     /* draw status first so it can be overdrawn by tags later */
-    if (m == selmon) { /* status is only drawn on selected monitor */
+    if (m == statmon) { /* status is only drawn on user-defined status monitor */
         tw = statusw = m->ww - drawstatusbar(m, bh, stext);
     }
 
@@ -2190,7 +2190,7 @@ updategeom(void)
                 else
                     mons = createmon();
             }
-            for (i = 0, m = mons; i < nn && m; m = m->next, i++)
+            for (i = 0, m = mons; i < nn && m; m = m->next, i++){
                 if (i >= n
                 || unique[i].x_org != m->mx || unique[i].y_org != m->my
                 || unique[i].width != m->mw || unique[i].height != m->mh)
@@ -2203,6 +2203,10 @@ updategeom(void)
                     m->mh = m->wh = unique[i].height;
                     updatebarpos(m);
                 }
+                if(i == statmonval)
+                    statmon = m;
+            }
+
         } else { /* less monitors available nn < n */
             for (i = nn; i < n; i++) {
                 for (m = mons; m && m->next; m = m->next);
@@ -2216,6 +2220,8 @@ updategeom(void)
                 }
                 if (m == selmon)
                     selmon = mons;
+                if (m == statmon)
+                    statmon = mons;
                 cleanupmon(m);
             }
         }
@@ -2303,7 +2309,7 @@ updatestatus(void)
 {
     if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
         strcpy(stext, "dwm-"VERSION);
-    drawbar(selmon);
+    drawbar(statmon);
 }
 
 void
